plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -19,18 +28,28 @@ var JobListWidget = View.extend({
         'click .g-job-trigger-link': function (e) {
             var cid = $(e.target).attr('cid');
             this.trigger('g:jobClicked', this.collection.get(cid));
+        },
+        'change select.g-page-size': function (e) {
+            this.collection.pageLimit = parseInt($(e.target).val());
+            this.pageSize = this.collection.pageLimit;
+            this.collection.fetch({}, true);
         }
     },
 
     initialize: function (settings) {
         var currentUser = getCurrentUser();
         this.showAllJobs = !!settings.allJobsMode;
         this.columns = settings.columns || this.columnEnum.COLUMN_ALL;
-        this.filter = settings.filter || {
-            userId: currentUser ? currentUser.id : null
-        };
-        this.typeFilter = {};
-        this.statusFilter = {};
+        this.userId = (settings.filter && !settings.allJobsMode) ? (settings.filter.userId ? settings.filter.userId : currentUser.id) : null;
+        this.typeFilter = null;
+        this.statusFilter = null;
+        this.phasesFilter = JobStatus.getAllStatus().reduce((obj, status) => {
+            obj[status.text] = true;
+            return obj;
+        }, {});
+
+        this.pageSizes = [25, 50, 100, 250, 500, 1000];
+        this.pageSize = 25;
```

Comment:
Do you suggestion on how to handle for the job/all situation?

This logic for page limit is simple enough to reuse what's in the collection.
Overall, I think I have a tendency to keep view layer and model layer away from each other. I mean, the pug doesn't need to know if there is one collection or two collections, the pug only need to know there is a page limit, the logic is kept within the View Model.